src/client/components/LandingView.react.js

```diff
@@ -17,38 +17,50 @@ type Props = {
 
 export default function LandingView(props: Props): React.Node {
   const { userId, setUserId, conn, setRoomState, roomId, setRoomId } = props;
+  const [errorMessage, setErrorMessage] = useState('');
+
+  function validateUsername(userId: string): boolean {
+    if (userId === '') {
+      setErrorMessage('Please choose a username!');
+      return false;
+    }
+    return true;
+  }
+
+  async function onCreateClick(): void {
+    console.log(`Create New request from userId: ${userId}`);
+    if (validateUsername(userId)) {
+      const roomState = await createRoom(conn, userId);
+      setRoomState(roomState);
+    }
+  }
+
+  async function onJoinClick(): void {
+    console.log(`Join Existing request from ${userId} for room ${roomId}`);
+    if (validateUsername(userId)) {
+      const response = await joinRoom(conn, userId, roomId);
+      if (response.error) {
+        console.log(`Join Existing failed: ${response.error}`);
+        setErrorMessage(response.error);
+        return;
+      }
+      setRoomState(response.roomState);
```

Comment:
i remember we discussed about maybe `reject`ing in the promise and doing a `try/catch` here, but i like this style more! it is less cluttered and requires less nesting